### DISCUSSION TITLE
NUnit3 doesn't use the .config file when run tests

### DISCUSSION THREAD

**Discussion by alexpantyukhin** [cake-build/cake#4223] [Q&A] [answered]
Hello!
I have a .Net project. 
It's for multiple frameworks:
```xml
    <TargetFrameworks>net48;net6.0</TargetFrameworks>
```

I run the following in the cake script:

```csharp
  var nunit3Settings = new NUnit3Settings 
        {
          NoResults = false,
          WorkingDirectory = bin,
          Results = new List<NUnit3Result>() { 
            new NUnit3Result() 
            {
              FileName = $"{output.ToString()}/{outputFileName}",
              Transform = "nunit3-junit.xslt",
            },
          },
        };

  NUnit3(nunitTestFilesPattern, nunit3Settings);
```

It works very well for the net48. But when run for net6.0 the code inside tests can't read the connection string.
I checked that the <testdllname>.config file is inside the bin folder. But It's not catched but test runner.

**Comment by nils-a** [comment]
At a first glance, this does not seem to be an inherent problem with the *Cake -> NUnit.Console -> net6.0-Project* "pipelene".

Are you sure it is not your test that is faulty?

Could you share a minimal demo that reproduces your problem?

**Comment by alexpantyukhin** [comment]
@nils-a  I have created a small demo https://github.com/alexpantyukhin/TestCake . There is a list of [outputs](https://github.com/alexpantyukhin/TestCake/tree/main/build/output) which contains results of tests. The rider works well on it, but when I use the Cake NUnit3

**Reply by devlead** [reply, marked as answer]
Could it be that rider uses .NET SDK dotnet test and not .NET Framework NUnit command line runner?
If so you could try the DotNetTest alias instead.

**Reply by alexpantyukhin** [reply]
Yes, DotNetTest uses sync results for net48 and net6.0. But it doesn't have settings configs for outputs and xslt transformations as I found for now.

**Reply by alexpantyukhin** [reply]
It seems that it has totally different values of `ConfigurationManager.OpenExeConfiguration(ConfigurationUserLevel.None).FilePath` for net48 and net6.0 . And it points to the `NUnit.ConsoleRunner.3.16.3` config in the second case.

**Comment by alexpantyukhin** [comment]
Thank you all. I think the issue is clear. It seems the problem is inside the nunit3 console. Another variant is to use DotNetTest as @devlead suggested,
